cpp/src/Ice/ServantManager.cpp

```diff
@@ -120,15 +120,15 @@ IceInternal::ServantManager::removeServant(const Identity& ident, const string&
     return servant;
 }
 
-shared_ptr<Object>
+ObjectPtr
 IceInternal::ServantManager::removeDefaultServant(const string& category)
 {
     //
     // We return the removed servant to avoid releasing the last reference count
     // with *this locked. We don't want to run user code, such as the servant
     // destructor, with an internal Ice mutex locked.
     //
-    shared_ptr<Object> servant = 0;
+    ObjectPtr servant = 0;
```

Comment:
```suggestion
    ObjectPtr servant = nullptr;
```

Or

`ObjectPtr servant {};`?

Not sure what is preferred.

Reply:
The preferred syntax is simply:
```cpp
ObjectPtr servant; // construct with default constructor
```